meraki/rest_session.py

```diff
@@ -6,12 +6,52 @@
 from datetime import datetime
 
 import requests
+from requests.utils import to_key_val_list
+from requests.compat import basestring, urlencode
 
 from meraki.__init__ import __version__
 from meraki.common import *
 from meraki.config import *
 
 
+def encode_params(_, data):
+    """Encode parameters in a piece of data.
+
+    Will successfully encode parameters when passed as a dict or a list of
+    2-tuples. Order is retained if data is a list of 2-tuples but arbitrary
+    if parameters are supplied as a dict.
+    """
+    if isinstance(data, (str, bytes)):
```

Comment:
Since we're re-working a key part of the `requests` library, it'd be helpful to see more documentation (comments) in this method. E.g.,

1. One comment per conditional and/or loop
2. One comment per result.append action
3. One comment per return